fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -2507,6 +2513,44 @@ ACTOR Future<Void> reportTxnTagCommitCost(UID myID,
 	}
 }
 
+// Get the list of tenants that are over the storage quota from the data distributor for quota enforcement.
+ACTOR Future<Void> monitorTenantsOverStorageQuota(UID myID,
+                                                  Reference<AsyncVar<ServerDBInfo> const> db,
+                                                  ProxyCommitData* commitData) {
+	state Future<Void> nextRequestTimer = Never();
+	state Future<TenantsOverStorageQuotaReply> nextReply = Never();
+	if (db->get().distributor.present())
+		nextRequestTimer = Void();
+	loop choose {
+		when(wait(db->onChange())) {
+			if (db->get().distributor.present()) {
+				CODE_PROBE(true, "DataDistributor changed during monitorTenantsOverStorageQuota");
+				TraceEvent("DataDistributorChanged", myID).detail("DDID", db->get().distributor.get().id());
```

Comment:
This case doesn't necessarily mean that the data distributor changed. It could be that `db` changed for some other reason, and the `distributor` is still the same.